### ISSUE TITLE
[Dashboard] [Home] Welcome Module

### ISSUE THREAD

**Issue by alonsogarciapablo** [CartoDB/cartodb#14527]
The original idea was to have 4 different states:

1. First time using the new dashboard user.
2. User with a notification from admin.
3. User with other notifications (eg: a product update)
4. Returning user with no notifications.

Designs are available here: https://projects.invisionapp.com/share/J7OPHL18HUR#/screens/336549667

Some of these states are easier than other.

## 1. First time using the new dashboard user.

The `/me` endpoint has some flags (eg `show_builder_activated_message`) to determine whether some messages should be displayed or not. I guess we would need to do something similar so that the frontend can know if the user is seeing the dashboard for the first time. What do you think @gonzaloriestra?

The problem I see with this is that someone might signup and not pay attention to the welcome message the first time. If we provide a link with some useful info about the new Dashboard, they might not see it. Perhaps we should show it until the user has performed some action (eg: click on the link to learn more or discard this "notification"). @kukukaka what do you think?

## 2. User with a notification from admin.

It should be easy to detect when there's one or more unread notifications and invite the user to view all notifications.

## 3. User with other notifications (eg: a product update).

The main challenges here are:
- Deciding which notifications we will display to users. Right now we're sending [emails to users](https://docs.google.com/spreadsheets/d/1j5Ke5UtUjKv5TmY1pwKIDnnhhIYwqoFd-E0esIYUQC8/edit#gid=0) based on some actions / account usage that could be helpful in the UI as well.
- We don't a system in place to do this. Ideally we would want to send notifications to users and make sure they see them / discard them. Right now only admins can send notifications to users.

An idea here is to push these notifications as any other notification instead of using the welcome module.

## 4. Returning user with no notifications.

Easy one too. If the user has already worked with the new dashboard and has no notifications, we display and compacted view of the welcome message.

---

In summary, 3. requires more work and it's kind of blocking this so I suggest we go with 1., 2. and .4 until we have a more clear idea of how we want to notify users of important things happening on their accounts.

@kukukaka  what do you think?

**Comment by gonzaloriestra** [comment]
Regarding 1:

Yes, I think we could add a new flag to the user model and update it after the first call of `/me`. In case we want to do it after a user action, we would have to call another endpoint to update it also.

But I have another proposal for the problem of not paying attention the first time: we could show the standard home view, but sending a "Welcome" notification when the user is created so that it appears there the first time and it can also be read later.

**Comment by alonsogarciapablo** [comment]
Thanks @gonzaloriestra.

>  In case we want to do it after a user action, we would have to call another endpoint to update it also.

Is there an endpoint to do this (eg: update some attribute of the user)? Thanks

**Comment by gonzaloriestra** [comment]
Yes, we use the same endpoint (/me), but with `PUT`.

**Comment by alonsogarciapablo** [comment]
Good to know! Thank you @gonzaloriestra!

**Comment by alonsogarciapablo** [comment]
After some discussion about this with @kukukaka we will start by only having three states:

## 1. First time using the new dashboard user

![image](https://user-images.githubusercontent.com/390398/50169711-69158200-02ee-11e9-9c94-33df13e7d83d.png)

## 2. Standard Welcome

![image](https://user-images.githubusercontent.com/390398/50169730-73d01700-02ee-11e9-8483-fe3e2628bc89.png)

## 3. Standard Welcome (During Trial)

![image](https://user-images.githubusercontent.com/390398/50169773-8cd8c800-02ee-11e9-9dee-8cb5530617a4.png)

**Comment by gonzaloriestra** [comment]
@alonsogarciapablo ok, so we need a new flag for the `/me` endpoint and it will be automatically disabled after the first call, right?

**Comment by kukukaka** [comment]
Sounds good!

Just a small detail I just commented with @jesusbotella the welcome messages are not fixed postion on scroll (the main nav with the blue background is).

**Comment by alonsogarciapablo** [comment]
@kukukaka we have to revisit the message / actions we show to users when they first see the Dashboard. There different types of users:

1. FREE user
2. 30-day trial
3. PRO user
4. Organization admin
5. Organization user

Here's the last design:

![image](https://user-images.githubusercontent.com/390398/50218787-6ddd4300-038d-11e9-8970-1802cc974999.png)

Here's a suggestion for the different user types described above:

**Your branch new CARTO Dashboard is ready! Take a look around and play with it or [read more about it on our blog](?).**

Remember that you're currently...

1. ... on the FREE plan.
2. ... on the 30-day personal trial.
3. ... on the Professional plan.
4. ... the admin of the ORGANIZATION_NAME organization.
5. ... a member of the ORGANIZATION_NAME organization.

What do you want to do next?

1. Create a map / Upload some data
2. Create a map / Upload some data
3. Create a map / Upload some data
4. Manage my organization
5. Create a map / Upload some data / Contact the admin

@kukukaka what do you think?

**Comment by jesusbotella** [comment]
Inspecting `/me` endpoint, I saw one property named `is_first_time_viewing_dashboard`. Maybe that's the property we are looking for ;)

But unfortunately I couldn't find any property to know when the trial finishes. :(

**Comment by gonzaloriestra** [comment]
Good news! Sorry, I should have checked it...

And we do have a property for the trial: `trial_ends_at` (under `user_data`).

**Comment by jesusbotella** [comment]
I have one doubt about the actions for the different type of users. Should we put the same actions no matter if it is the first time they visit the new dashboard or not?
